### PR TITLE
Update AWS CLI to v2 and remove Python27

### PR DESCRIPTION
**Pull request by bnurzhanov**
*Description of changes:*
Update user data to:
- remove installation of Python 2.7 PIP package (Python 2.7 is deprecated)
- install AWS CLI v2 instead of AWS CLI v1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bnurzhanov** [comment]
I tested this change by updating launch template in my account and then starting up an EC2 instance using updated template, logging into the instance and confirming correct version of AWS CLI:
```
aws --version
aws-cli/2.0.3 Python/3.7.3 Linux/4.14.158-101.185.amzn1.x86_64 botocore/2.0.0dev7
```

File: docs/core-env/create-custom-compute-resources.md [hunk 1]
```diff
@@ -37,23 +37,25 @@ Content-Type: text/cloud-config; charset="us-ascii"
...
 - jq
 - btrfs-progs
-- python27-pip
 - sed
 - wget
+- unzip
 # add more package names here if you need them
 
 runcmd:
-- pip install -U awscli boto3
```
Comment (wleepang) [inline docs/core-env/create-custom-compute-resources.md]: Need to keep `python-pip` and `boto3` as it's a dependency for ebs-autoscale.  That said, might be able to move that there.

Comment (wleepang) [reply docs/core-env/create-custom-compute-resources.md]: Pushed an update to awslabs/amazon-ebs-autoscale that removes the dependency on python, so this is fine now.

File: docs/core-env/create-custom-compute-resources.md [hunk 2]
```diff
@@ -37,23 +37,25 @@ Content-Type: text/cloud-config; charset="us-ascii"
...
+- curl -s "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "/tmp/awscliv2.zip"
+- unzip -q /tmp/awscliv2.zip -d /tmp
+- sudo /tmp/aws/install
```
[the remark this answers is not in the document]
Comment (bnurzhanov) [reply docs/core-env/create-custom-compute-resources.md]: updated in the latest commit

**Comment by bnurzhanov** [comment]
Pushed new commit, fixing IAM permissions

**Pull request merged**